config/default.yaml

```diff
@@ -163,3 +163,5 @@ ecosystem:
     store: https://store.screwdriver.cd
     # Badge service (needs to add a status and color)
     badges: https://img.shields.io/badge/build-{{status}}-{{color}}.svg
+    # Default registry to pull build containers from. Uses Docker Hub if nothing/empty string is provided
+    dockerRegistry: ""
```

Comment:
I think the main issue captures it well (https://github.com/screwdriver-cd/screwdriver/issues/472), but I'll see if I can clarify it.

If you have an image `screwdrivercd/screwdriver:latest`, Docker needs to know which registry to pull the image from. When no registry is specified (like in this example), it will default to using `registry.hub.docker.com`. So when you say `screwdrivercd/screwdriver:latest`, it's implied that the registry is Docker Hub.

If you want to pull the image from a different registry, you would add the prefix to it, like `private.registry.com/screwdrivercd/screwdriver:latest`. Then, when the build starts, it will pull the image from `private.registry.com` instead of Docker Hub.

The feature this introduces is that if this setting is not configured, it will use the default Docker behavior (e.g., pull images from the implied registry). If you want all builds to pull images from a non-Docker Hub registry by default, you would configure this to your specific registry.